pydoll/browser/page.py

```diff
@@ -328,20 +330,31 @@ async def callback_wrapper(event):
             event_name, function_to_register, temporary
         )
 
-    async def execute_js_script(self, script: str) -> dict:
+    async def execute_script(self, script: str, element: WebElement = None):
         """
         Executes a JavaScript script in the page.
+        If an element is provided, the script will be executed in the context
+        of that element. To provide the element context, use the 'argument'
+        keyword in the script.
+
+        Examples:
+        ```python
+        await page.execute_script('argument.click()', element)
+        await page.execute_script('argument.value = "Hello, World!"', element)
+        ```
 
         Args:
             script (str): The JavaScript script to execute.
-
-        Returns:
-            dict: The result of the JavaScript script execution.
         """
-        command = {
-            'method': 'Runtime.evaluate',
-            'params': {'expression': script, 'returnByValue': True},
-        }
+        if element:
+            script = script.replace('argument', 'this')
+            script = f'function(){{ {script} }}'
+            object_id = element._node['objectId']
```

Comment:
esse objetoID não do hmtl né? é um hash do js?

Reply:
não, isso é retornado pelo CDP quando tu faz um find_element, daí ele usa esse objectId como ponto de partida pra executar o script em cima dele. Assim o 'this' vai se referir ao elemento, por isso tem aquele replace